src/ebonite/ext/prebuild.py

```diff
@@ -0,0 +1,62 @@
+import os
+import re
+import shutil
+import sys
+import tempfile
+
+from jinja2 import Environment, FileSystemLoader
+
+from ebonite.build.docker import (RemoteDockerRegistry, create_docker_client, image_exists_at_dockerhub,
+                                  login_to_registry, repository_tags_at_dockerhub)
+from ebonite.utils.log import logger
+
+
+def prebuild_image(prebuild_path, name_template, python_version, *, push=False):
+    tag = name_template.format(python_version)
+    if image_exists_at_dockerhub(tag):
+        logger.info('Skipped building image %s: already exists', tag)
+        return
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        context_dir = os.path.join(tmpdir, 'context')
+        logger.info('Building image %s on %s ...', tag, context_dir)
+
+        shutil.copytree(prebuild_path, context_dir)
+        _generate_dockerfile(context_dir, python_version)
+
+        try:
+            with create_docker_client() as client:
+                client.images.build(path=context_dir, tag=tag, rm=True)
+
+            logger.info('Successfully built image %s', tag)
+        except Exception as e:
+            logger.error('Failed to build image %s: %s', tag, e)
+            return
+
+    if push:
+        with create_docker_client() as client:
+            login_to_registry(client, RemoteDockerRegistry())
+            client.images.push(tag)
+
+
+def _generate_dockerfile(context_dir, python_version):
+    j2 = Environment(loader=FileSystemLoader([context_dir]))
+    docker_tmpl = j2.get_template('Dockerfile.j2')
+    with open(os.path.join(context_dir, 'Dockerfile'), 'w') as df:
+        df.write(docker_tmpl.render({'python_version': python_version}))
+
+
+PY_BLACK_LIST = {'^2\\.', '^3\\.[0-5]\\.', '^3\\.6\\.[0-4]'}
+
+
+def prebuild_missing_images(prebuild_path, name_template):
```

Comment:
this code and it's helper functions feel like they dont belong to ebonite codebase =\ anyway there's gonna be something that is gonna run it on schedule, maybe move it there completely?

Reply:
Yep, I planned to detach this code completely from Ebonite in future. But we should be aware of potentially high maintenance burden (dealing with several repos) in this case. That's why for now I decided to leave it here but already make it as Ebonite-agnostic as possible.